src/charm.py

```diff
@@ -303,6 +302,19 @@ def logs_rules(self) -> Dict[str, Any]:
         """Return a list of logging rules."""
         return self._cos.logs_alerts
 
+    @property
+    def requested_receivers(self) -> set:
+        """Return a list of requested tracing receivers."""
+        protocols = self._cos.requested_protocols()
+        protocols.update(
+            [
+                receiver
+                for receiver in get_args(ReceiverProtocol)
+                if self.config.get(f"always_enable_{receiver}") is True
```

Comment:
no need for `is true`, since `get(x)` defaults to the falsy `None`.

putting together the comments above, how about:

```python
protocols.update(
    filter(self.config.get(f"always_enable_{receiver}") for receiver in get_args(ReceiverProtocol))
)
```

Reply:
Went almost all the way, without `filter(` though.